python/ray/ml/train/data_parallel_trainer.py

```diff
@@ -1,16 +1,36 @@
+import inspect
 import logging
-from typing import Dict, Callable, Optional, Union
+import os
+from typing import Dict, Callable, Optional, Union, Type
 
+import ray
+from ray import tune
+from ray.ml.constants import TRAIN_DATASET_KEY
 from ray.ml.trainer import Trainer
-from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.config import ScalingConfig, RunConfig, ScalingConfigDataClass
 from ray.ml.trainer import GenDataset
 from ray.ml.preprocessor import Preprocessor
 from ray.ml.checkpoint import Checkpoint
-from ray.train import BackendConfig
+from ray.train import BackendConfig, TrainingIterator
+from ray.train.backend import BackendExecutor
+from ray.train.checkpoint import MLTuneCheckpointManager
+from ray.train.constants import (
+    ENABLE_DETAILED_AUTOFILLED_METRICS_ENV,
+    ENABLE_SHARE_CUDA_VISIBLE_DEVICES_ENV,
+)
+from ray.train.utils import construct_train_func
+from ray.tune import Trainable
 from ray.util.annotations import DeveloperAPI
 
 logger = logging.getLogger(__name__)
 
+# The environment variables that need to be propagated from the driver to the
+# `BackendExecutor` actor via runtime env.
+_BACKEND_ENV_VARS = {
+    ENABLE_DETAILED_AUTOFILLED_METRICS_ENV,
+    ENABLE_SHARE_CUDA_VISIBLE_DEVICES_ENV,
+}
+
```

Comment:
Long-term I believe Runtime Environments are getting rid of merge semantics, so this would save us some overhead! And this would generally be more consistent with how users should think about environment variables with Ray applications.

Reply:
> Isn't this what this PR is doing?

Ah woops, I meant explicitly in cluster config!

> Is that something we want to require users to do?

I actually do think this is fine given that the way we've been using environment variables is as "experimental" flags. If we find a real need for the user to configure these we should add them as explicit args.

Reply:
My opinion is we can remove all this logic for the time being and if there's a need that user-defined runtime environments don't solve we can add a comprehensive solution in.

Reply:
Ok removed